lib/interface.js

```diff
@@ -1,82 +1,87 @@
-  this._options = this._options || [];
-  this._options.push(_.clone(opts) || {});
-  return this;
-};
+  // Create a new instance of the `Runner`, passing in the current object.
+  Target.prototype.then = function(/* onFulfilled, onRejected */) {
+    var result = this.client.runner(this).run()
+    return result.then.apply(result, arguments);
+  };
 
-// Sets an explicit "connnection" we wish to use for this query.
-Target.prototype.connection = function(connection) {
-  this._connection = connection;
-  return this;
-};
+  // Add additional "options" to the builder. Typically used for client specific
+  // items, like the `mysql` and `sqlite3` drivers.
+  Target.prototype.options = function(opts) {
+    this._options = this._options || [];
+    this._options.push(_.clone(opts) || {});
+    return this;
+  };
 
-// Set a debug flag for the current schema query stack.
-Target.prototype.debug = function(enabled) {
-  assert(!arguments.length || typeof enabled === 'boolean', 'debug requires a boolean');
-  this._debug = arguments.length ? enabled : true;
-  return this;
-};
+  // Sets an explicit "connnection" we wish to use for this query.
+  Target.prototype.connection = function(connection) {
+    this._connection = connection;
+    return this;
+  };
 
-// Set the transaction object for this query.
-Target.prototype.transacting = function(t) {
-  this._transacting = t;
-  return this;
-};
+  // Set a debug flag for the current schema query stack.
+  Target.prototype.debug = function(enabled) {
+    assert(!arguments.length || typeof enabled === 'boolean', 'debug requires a boolean');
+    this._debug = arguments.length ? enabled : true;
+    return this;
+  };
 
-// Initializes a stream.
-Target.prototype.stream = function(options) {
-  var Runner = this.client.Runner;
-  return new Runner(this).stream(options);
-};
+  // Set the transaction object for this query.
+  Target.prototype.transacting = function(t) {
+    if (t && t.client) {
+      if (!t.client.transacting) {
+        helpers.warn('Invalid transaction value: ' + t.client)
+      } else {
+        this.client = t.client
+      }
+    }
+    return this;
+  };
 
-// Initialize a stream & pipe automatically.
-Target.prototype.pipe = function(writable) {
-  var Runner = this.client.Runner;
-  return new Runner(this).pipe(writable);
-};
+  // Initializes a stream.
+  Target.prototype.stream = function(options) {
+    return this.client.runner(this).stream(options);
+  };
 
-// Creates a method which "coerces" to a promise, by calling a
-// "then" method on the current `Target`
-_.each(['bind', 'catch', 'spread', 'otherwise', 'map', 'reduce', 'tap', 'thenReturn',
-  'return', 'yield', 'ensure', 'nodeify', 'exec'], function(method) {
-  Target.prototype[method] = function() {
-    var then = this.then();
-    return then[method].apply(then, arguments);
+  // Initialize a stream & pipe automatically.
+  Target.prototype.pipe = function(writable) {
```

Comment:
pipe actually takes a second argument with options, probably just want to pass that along verbatim to the other pipe